### ISSUE TITLE
ValueError: could not broadcast input array from shape (124,4) into shape (50,4)

### ISSUE THREAD

**Issue by eonurk** [NVIDIA/DIGITS#1235]
Here is the last couple of line out my caffe_output.log file;
```
File "/home/serkan/Desktop/caffe-caffe-0.15/python/caffe/layers/detectnet/clustering.py", line 66, in forward
bbox = cluster(self, data0, bottom[1].data)
File "/home/serkan/Desktop/caffe-caffe-0.15/python/caffe/layers/detectnet/clustering.py", line 232, in cluster
boxes[i, 0:r, 0:c] = boxes_cur_image
ValueError: could not broadcast input array from shape (124,4) into shape (50,4)
```
What might the problem?

**Comment by lukeyeager** [comment]
See https://github.com/NVIDIA/caffe/issues/228, https://github.com/NVIDIA/caffe/pull/238.

You need to change the values of `MAX_BOXES` [here](https://github.com/NVIDIA/caffe/blob/v0.15.13/python/caffe/layers/detectnet/clustering.py#L8) and [here](https://github.com/NVIDIA/caffe/blob/v0.15.13/python/caffe/layers/detectnet/mean_ap.py#L5) to something larger than 124 - maybe 150?

**Comment by eonurk** [comment]
Thank you very much @lukeyeager, I passed that error. However, now I'm facing an out of memory issue. Batch size&accumalation are 1&1 and I have a gtx650m. 
[caffe_output(1).log.zip](https://github.com/NVIDIA/DIGITS/files/567164/caffe_output.1.log.zip)

**Comment by lukeyeager** [comment]
You're not going to be able to train DetectNet with a 1GB GPU, sorry. I've done it with a 4GB GPU, but it's much slower than with a 12GB GPU.

> NOTE: this instance of DetectNet requires at least 12GB of GPU memory. If you have less memory on your GPU[s], you may want to decrease the batch size. On a 4GB card, you can set the batch size to 2 and the batch accumulation to 5, for an effective batch of 10, and that should fit on your card.
https://github.com/NVIDIA/DIGITS/blob/digits-5.0/examples/object-detection/README.md#model-creation

**Comment by eonurk** [comment]
It's 2 but I don't think it does not work too :( thanks anyways.

**Comment by lukeyeager** [comment]
Closing since your original question was answered.

**Comment by caleale90** [comment]
Hi, I am facing the same error of this issue.
I followed the @lukeyeager instruction and changed the variable MAX_BOXES = 150 in both files clustering.py and mean_ap.py. Then I recompiled the entire caffe library, but I obtain the same error. 
This is the last part of my caffe_output.log file:

I0616 09:35:50.296716  5248 sgd_solver.cpp:106] Iteration 20000, lr = 0.0001
I0616 09:40:43.958341  5248 solver.cpp:362] Iteration 20759, Testing net (#0)
I0616 09:40:43.958425  5248 net.cpp:723] Ignoring source layer train-data
I0616 09:40:43.958427  5248 net.cpp:723] Ignoring source layer train-label
I0616 09:40:43.958429  5248 net.cpp:723] Ignoring source layer train-transform
Traceback (most recent call last):
File "/usr/lib/python2.7/dist-packages/caffe/layers/detectnet/clustering.py", line 133, in forward
bbox = cluster(self, data0, bottom[1].data)
File "/usr/lib/python2.7/dist-packages/caffe/layers/detectnet/clustering.py", line 232, in cluster
boxes[i, 0:r, 0:c] = boxes_cur_image
ValueError: could not broadcast input array from shape (51,5) into shape (50,5)

**Comment by gfjiyue** [comment]
If I run with docker, how to change the value of max_boxes? @lukeyeager

**Comment by apollack11** [comment]
@gfjiyue This is a bit hacky, but I believe it worked.. I ran `sudo find / -name clustering.py` and  `sudo find / -name mean_ap.py` to see where the files were located in docker. I was able to find both files and change MAX_BOXES in both of them. I then restarted the docker container with `sudo docker restart [container_id]`. Everything seems to be working now even if I have larger labels. Not sure if these changes will persist with a restart of the machine though
